src/providers/debug_config_provider.ts

```diff
@@ -129,6 +107,35 @@ export class DebugConfigProvider implements DebugConfigurationProvider {
 		return debugConfig;
 	}
 
+	private guessBestEntryPoint(openFile: string, workspaceFolder: WorkspaceFolder | undefined): string {
+
+		if (isTestFile(openFile) || isInsideFolderNamed(openFile, "bin") || isInsideFolderNamed(openFile, "tool")) {
+			return openFile;
+		} else {
+			// Use the open file as a clue to find the best project root, then search from there.
+			const commonLaunchPaths = [
+				path.join(fsPath(workspaceFolder.uri), "lib", "main.dart"),
+				path.join(fsPath(workspaceFolder.uri), "bin", "main.dart"),
+			];
+			for (const launchPath of commonLaunchPaths) {
+				if (fs.existsSync(launchPath)) {
+					return launchPath;
+				}
+			}
+		}
+	}
+
+	private async getObservatoryUri(): Promise<string> {
+		let userInput = await vs.window.showInputBox({ prompt: "Enter Observatory address. This can be a full URL, or just a port for localhost." });
+
+		// If the input is just a number, treat is as a localhost port.
+		if (userInput && /^\s*[0-9]+\s*$/.exec(userInput)) {
+			userInput = `http://127.0.0.1:${userInput}`;
```

Comment:
This lost the trim(), so if there's whitespace at the start it'll create an invalid URL.